stretch_core/stretch_core/joint_trajectory_server.py

```diff
@@ -341,10 +341,34 @@ def error_callback(self, goal_handle, error_code, error_str):
         return result
 
     def cancel_cb(self, goal_handle):
-        """Accept or reject a client request to cancel an action."""
-        self.node.get_logger().info("{0} joint_traj action: received cancel request".format(self.node.node_name))
-        # self.node.robot.stop_trajectory()
-        return CancelResponse.ACCEPT # Accepting cancel request
+        """Accept or reject a client request to cancel an action.
+        """
+        self.node.robot_mode_rwlock.acquire_read()
+        if self.node.robot_mode in ['position','navigation']:
+            self.node.robot_mode_rwlock.release_read()
+            # Stop all joints
+            self.node.robot.base.left_wheel.enable_safety()
```

Comment:
In 'position' mode the base can't be moved, right? Perhaps the wheels only need to be stopped in 'navigation' mode?

Reply:
The base can be moved in 'position' mode through the FJT action using "translate_mobile_base" and "rotate_mobile_base" as joint names. They're mutually exclusive ofc.